cpp/src/arrow/compute/api_scalar.h

```diff
@@ -272,12 +272,27 @@ struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
   constexpr static char const kTypeName[] = "DayOfWeekOptions";
   static DayOfWeekOptions Defaults() { return DayOfWeekOptions{}; }
 
-  /// Number days from 1 if true and from 0 if false
+  /// Number from 1 if true and from 0 if false
```

Comment:
I think it's best to put back `days` in the `DayOfWeekOptions` docs since we're now using a different options class for this
```suggestion
  /// Number days from 1 if true and from 0 if false
```